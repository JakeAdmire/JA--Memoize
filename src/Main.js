import React, { Component } from 'react';
import Card from './Card.js';

export default class Main extends Component {
  constructor(props) {
    super(props);
    this.state = { cardNumber: 0 };
  } 
  showNextCard = () => {
    if ((this.state.cardNumber + 1) < this.props.cards.length) {
      this.setState({cardNumber: this.state.cardNumber += 1});
      this.props.resetState({cardNumber: this.state.cardNumber++});
    }
  }
  showPrevCard = () => {
    if (this.state.cardNumber !== 0) {
      this.setState({cardNumber: this.state.cardNumber -= 1});
      this.props.resetState({cardNumber: this.state.cardNumber--});
    }
  }
  generateCards = () => {
    const cards = this.props.cards;
    const cardsList = cards.map((card) => {
      return (
        <Card 
          frontContent={card}
          backContent={card.solutionPrototype}
          cardNumber={this.state.cardNumber}
          nextCard={this.showNextCard}
          prevCard={this.showPrevCard}
          answers={this.props.answers}
          correctAnswer={this.props.correctAnswer}
        />
      );
    })
   return(cardsList[this.state.cardNumber]); 
  }
  render() {
    return (
      <div className="main">
        {this.generateCards()}
      </div>
    )
  }
}