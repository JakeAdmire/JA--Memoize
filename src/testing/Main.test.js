import React from 'react';
import ReactDOM from 'react-dom';
import Main from '../Main';
import { shallow } from 'enzyme';

const cardsMock = [{}, {}, {}]
const resetStateMock = jest.fn();

describe('Main', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Main cards={cardsMock}
            resetState={resetStateMock}
            />
    );
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(wrapper.state()).toEqual({ 
      cardNumber: 0
     });
  })

  // it('should increase card number when invoked', () => {
  //   expect(wrapper.state('cardNumber')).toEqual(0);
  //   wrapper.instance().increaseCardNumber();
  //   expect(wrapper.state('cardNumber')).toEqual(1);
  // })

  it('should move to the next card when invoked', () => {
    expect(wrapper.state('cardNumber')).toEqual(0);
    wrapper.instance().showNextCard();
    expect(wrapper.state('cardNumber')).toEqual(2);
  })

  it('should move to the previous card when invoked', () => {
    wrapper.instance().showNextCard();
    expect(wrapper.state('cardNumber')).toEqual(2);
    wrapper.instance().showPrevCard();
    expect(wrapper.state('cardNumber')).toEqual(0);
  })

})