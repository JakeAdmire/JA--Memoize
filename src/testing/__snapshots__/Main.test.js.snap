// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    cards={
      Array [
        Object {},
        Object {},
        Object {},
      ]
    }
    resetState={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Card
        cardNumber={0}
        frontContent={Object {}}
        increaseCardNumber={[Function]}
        nextCard={[Function]}
        prevCard={[Function]}
        resetState={[MockFunction]}
      />,
      "className": "main",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "answers": undefined,
        "backContent": undefined,
        "cardNumber": 0,
        "correctAnswer": undefined,
        "frontContent": Object {},
        "increaseCardNumber": [Function],
        "nextCard": [Function],
        "prevCard": [Function],
        "resetState": [MockFunction],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Card
          cardNumber={0}
          frontContent={Object {}}
          increaseCardNumber={[Function]}
          nextCard={[Function]}
          prevCard={[Function]}
          resetState={[MockFunction]}
        />,
        "className": "main",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "answers": undefined,
          "backContent": undefined,
          "cardNumber": 0,
          "correctAnswer": undefined,
          "frontContent": Object {},
          "increaseCardNumber": [Function],
          "nextCard": [Function],
          "prevCard": [Function],
          "resetState": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
